gems/smithy/lib/smithy/generators/client.rb

```diff
@@ -11,41 +11,67 @@ def initialize(plan)
         super
       end
 
-      # @return [Enumerator<String, String>] The file paths and their contents to generate.
       def generate
-        files = source_files
-        files.each do |file, content|
+        gem_files.each_with_object([]) do |(file, content), files|
           next if file == "lib/#{@gem_name}/customizations.rb" && should_skip_customizations?
 
           create_file file, content
+          files << file
         end
-        files
+      end
+
+      def source
+        source = source_files.map { |_file, content| content }.join("\n")
+        Object.module_eval(source)
+        Object.const_get(@plan.gem_namespace)
+      rescue Exception => e # rubocop:disable Lint/RescueException
+        puts "Error evaluating source:\n#{source}"
+        raise e
       end
 
       private
 
-      # rubocop:disable Metrics/AbcSize
-      def source_files
+      def gem_files
         Enumerator.new do |e|
           e.yield "#{@gem_name}.gemspec", Views::Client::Gemspec.new(@plan).render
           e.yield '.rubocop.yml', Views::Client::RubocopYml.new(@plan).render
 
-          e.yield "lib/#{@gem_name}.rb", Views::Client::Module.new(@plan).render
+          source_files.each { |file, content| e.yield file, content }
           e.yield "lib/#{@gem_name}/customizations.rb", Views::Client::Customizations.new.render
```

Comment:
This feels a little weird to me as a special case - why isn't it considered a source file?

Reply:
I don't think customizations would be a source file because the file persists as a written file in the gem. There would be no way to load a customization if it were not from a file.

Reply:
I guess I don't understand what a source file is then. I assume source => a ruby source file (ie, any file ending in .rb).

Reply:
It's source code. The source files are iterated just for their source. Do you have a better name?